contentprovider.go

```diff
@@ -261,41 +261,49 @@ func (nls newlines) atOffset(offset uint32) (lineNumber, lineStart, lineEnd int)
 		return nls.locs[n] >= offset
 	})
 
-	end := int(nls.fileSize)
-	if idx < len(nls.locs) {
-		end = int(nls.locs[idx])
+	start, end := nls.lineBounds(idx + 1)
+	return idx + 1, int(start), int(end)
+}
+
+// lineBounds returns the byte offsets of the start and end of the 1-based
+// lineNumber. The end offset is exclusive and will not contain the line-ending
+// newline. If the line number is out of range of the lines in the file, start
+// and end will be clamped to [0,fileSize].
+func (nls newlines) lineBounds(lineNumber int) (start, end uint32) {
+	// newlines[0] is the start of the 2nd line in data.
```

Comment:
s/newlines/nls.locs/

Reply:
and this is not true?
start of the 2nd line is ```nls.locs[0] + 1```, isn't it?

Reply:
Follow up here: https://github.com/sourcegraph/zoekt/pull/369